src/omv/ports/mimxrt/sensor.c

```diff
@@ -198,6 +198,11 @@ void sensor_sof_callback() {
         // Missed a few lines, reset buffer state and continue.
         buffer->reset_state = true;
     }
+    // Mimic gpio vsync callback.
+    if (sensor.vsync_callback) {
+        sensor.vsync_callback(sensor.hw_flags.vsync);
+        sensor.vsync_callback(!sensor.hw_flags.vsync);
+    }
```

Comment:
Yeah, I don't know if it's possible to share the GPIO pin on the IMX RT like the STM32 allows with the DCMI hardware. I tried the same code as the STM32 for doing a GPIO interrupt and it just disconnects the VSYNC line from the CSI hardware so that it doesn't work anymore.

So, I looked at the order of high/low interrupts generated by the STM32 VSYNC line and copied it to match, in the code above. This mimics what the STM32 does more or less. It gets called at the start of the frame in the SOF interrupt.

Reply:
Yes but doesn't get called twice in a row. Also you can try to just read the GPIO, if it doesn't work, try
```diff
- OMV_GPIO_DEFINE(DCMI_VSYNC, B1_13, CSI, 0)
+ OMV_GPIO_DEFINE(DCMI_VSYNC, B1_13, CSI, 1)
```

Reply:
The STM32 interrupt happens on the rising and falling edge of vsync. It happens once at the end of the frame and then at the start of the next. The code above mimics that given the vsync polarity. However, both events happen at the start of the frame versus one at the end of the frame and one at the start like for the STM32. Reading the gpio pin isn't needed as I need to generate two callbacks to match the STM32 driver and the vsync interrupt itself is triggered by the signal edge.

Reply:
Okay, I looked into this and found that omv_gpio_irq_handler is not hooked into anything right now on the IMX.

I tried adding that into the machine.c handler and then copying what was done for the STM32 and the interrupt doesn't trigger. Video streams fine though, no crashes.

Anyway, just going to remove vsync functionality for now. It can be added back again once the omv_gpio.c file's interrupt support is integrated and working.